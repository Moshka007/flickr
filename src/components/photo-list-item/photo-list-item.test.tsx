import React from "react";
import { render, screen } from "@testing-library/react";
import PhotoListItem from './photo-list-item'
import userEvent from "@testing-library/user-event";

const url: string = 'https://images.unsplash.com/photo-1508921912186-1d1a45ebb3c1?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8cGhvdG98ZW58MHx8MHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=600&q=60',
      tags: string[] = ['test', 'small'],
      id: number = 1,
      setF = jest.fn();

describe('Testing <PhotoListItem/>', () => {
    it('PhotoListItem have "bookMark it" (main page)', () => {
        render(<PhotoListItem url={url} tags={tags} id={id} bookMark={false} setBookMarkArray={setF}/>);
        expect(screen.getByText(/bookmark it/i)).toBeInTheDocument();
    })

    it('PhotoListItem have "Remove it" (bookmark page)', () => {
        render(<PhotoListItem url={url} tags={tags} id={id} bookMark={true} setBookMarkArray={setF}/>);
        expect(screen.getByText(/remove it/i)).toBeInTheDocument();
    })

    it('PhotoListItem have some tags', () => {
        render(<PhotoListItem url={url} tags={tags} id={id} bookMark={false} setBookMarkArray={setF}/>);
        expect(screen.getByText(/#test/i)).toBeInTheDocument();
        expect(screen.getByText(/#small/i)).toBeInTheDocument();
    })

    it('PhotoListItem have img', () => {
        render(<PhotoListItem url={url} tags={tags} id={id} bookMark={false} setBookMarkArray={setF}/>);
        expect(screen.getByRole('img')).toBeInTheDocument();
    })

    it('PholoListItem have rendered correctly', () => {
        const photoListItem = render(<PhotoListItem url={url} tags={tags} id={id} bookMark={false} setBookMarkArray={setF}/>);
        expect(photoListItem).toMatchSnapshot();
    })
})
describe('Tescting onClick buttons', () => {
    it('Testing onClick handler "bookmark it"',() => {
        render(<PhotoListItem url={url} tags={tags} id={id} bookMark={false} setBookMarkArray={setF}/>);
        userEvent.click(screen.getByRole('button'));

        expect(setF).toHaveBeenCalled();
    })

    it('Testing onClick handler "Remove it"',() => {
        render(<PhotoListItem url={url} tags={tags} id={id} bookMark={true} setBookMarkArray={setF}/>);
        userEvent.click(screen.getByRole('button'));

        expect(setF).toHaveBeenCalled();
    })
})
