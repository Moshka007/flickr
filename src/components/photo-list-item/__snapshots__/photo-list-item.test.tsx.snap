// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PhotoListItem/> PholoListItem have rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pt-5 col-md-4"
      >
        <div
          class="card"
          style="height: 100%;"
        >
          <div
            style="height: calc(100% - 6.5em);"
          >
            <img
              class="card-img-top"
              src="https://images.unsplash.com/photo-1508921912186-1d1a45ebb3c1?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8cGhvdG98ZW58MHx8MHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=600&q=60"
              style="height: 100%; width: 100%; padding: 1.5em 1em 0px 1em; object-fit: cover;"
            />
          </div>
          <div
            class="card-body"
          >
            <button
              class="btn btn-light"
              style="cursor: pointer; border: 1px solid black; opacity: 0.7;"
              type="button"
            >
              Bookmark it!
            </button>
            <p
              class="mt-2 p-1 pl-2 card-text"
              style="border: 1px solid black; opacity: 0.7; border-radius: 5px;"
            >
              #test 
              #small 
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pt-5 col-md-4"
    >
      <div
        class="card"
        style="height: 100%;"
      >
        <div
          style="height: calc(100% - 6.5em);"
        >
          <img
            class="card-img-top"
            src="https://images.unsplash.com/photo-1508921912186-1d1a45ebb3c1?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8cGhvdG98ZW58MHx8MHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=600&q=60"
            style="height: 100%; width: 100%; padding: 1.5em 1em 0px 1em; object-fit: cover;"
          />
        </div>
        <div
          class="card-body"
        >
          <button
            class="btn btn-light"
            style="cursor: pointer; border: 1px solid black; opacity: 0.7;"
            type="button"
          >
            Bookmark it!
          </button>
          <p
            class="mt-2 p-1 pl-2 card-text"
            style="border: 1px solid black; opacity: 0.7; border-radius: 5px;"
          >
            #test 
            #small 
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
